operator: fix bugs on reading configuration from config-map

aanm opened the pull request:
`cobra.OnInitialize` should be called directly in the init function.

Due refactoring, `cilium-agent --version` and `cilium-operator --version` no longer work, this PR also fixes this bug by rewriting the code.

Also, stop executing flag validation if flags are not set

Fixes: 67b147c3f273 ("operator: add ability to ready options from a config directory")
Signed-off-by: André Martins <andre@cilium.io>

Fixes https://github.com/cilium/cilium/issues/10384


<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/10520)
<!-- Reviewable:end -->

aanm committed:
operator: read options from config-map

cobra.OnInitialize should be called directly in the init function.

Fixes: 67b147c3f273 ("operator: add ability to ready options from a config directory")
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
operator/cilium: only print version with --version

The programs should be executed when running with `--version` and should
only print the version and terminate immediately.

Fixes: 67b147c3f273 ("operator: add ability to ready options from a config directory")
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
pkg/option: ignore warnings if flags are not set

Signed-off-by: André Martins <andre@cilium.io>

tklauser reviewed (commented):
LGTM in general.

Noticed that `operator` now has two `init` functions one of which (in `operator/flags.go) calls `cobra.OnInitialize(option.InitConfig...)` hooking up the handling of `--version` and the other (in `operator/main.go`) handles `--version` by itself. Would it be possible to somehow merge the two so we can a centralized handling of `--version` in `InitConfig`?

aanm committed:
test: delete cilium operator deployment

As cilium operator now relies in the config map changes, we need to
restart it the same way as cilium agent so that it can pick up new
changes from the cilium-config `config-map`.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
test: fix message with number of nodes running cilium

Fixes: 622dae393fd4 ("Test: Added Pre flight checks on ExpectCiliumReady")
Signed-off-by: André Martins <andre@cilium.io>

nebril commented on test/helpers/kubectl.go:
```diff
@@ -1493,6 +1493,43 @@ func (kub *Kubectl) GetPublicIface() (string, error) {
+
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out waiting to delete Cilium Operator: pods still remaining: %s", pods)
```
`pods` will be uninitialized here on first run, probably won't ever happen, but can we move whole `select` to the end of the loop body? I assume we want to run through the loop body at least once.

aanm replied:
@nebril I'm going to fix this in another PR as this PR is blocking some of my work.

nebril reviewed (changes requested):
One small comment, lgtm otherwise
